bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -265,36 +279,34 @@ public synchronized BigtableClusterAdminClient getClusterAdminClient() {
    * To create a pool, createPool has to be true and channel count has to be greater than one.
    * </p>
    */
-  protected Channel createChannel(final TransportOptions transportOptions, int channelCount) {
-
-    Preconditions.checkArgument(
-      transportOptions.getTransport() == TransportOptions.BigtableTransports.HTTP2_NETTY_TLS,
-      "Bigtable requires the NETTY_TLS transport.");
-    long timeoutMs = options.getTimeoutMs();
-
+  protected Channel createChannel(String hostString, int channelCount) throws IOException {
+    InetAddress host = getHost(hostString);
     Channel channels[] = new Channel[channelCount];
     for (int i = 0; i < channelCount; i++) {
-      ReconnectingChannel reconnectingChannel =
-          createReconnectingChannel(transportOptions, batchPool, timeoutMs);
+      ReconnectingChannel reconnectingChannel = createReconnectingChannel(host);
       clientCloseHandlers.add(reconnectingChannel);
       channels[i] = reconnectingChannel;
     }
-    Channel channel = new ChannelPool(channels);
-    return wrapChannel(channel);
+    return wrapChannel(new ChannelPool(channels));
   }
 
-  protected ReconnectingChannel createReconnectingChannel(
-      final TransportOptions transportOptions, final ExecutorService executor, long timeoutMs) {
-    return new ReconnectingChannel(timeoutMs, new ReconnectingChannel.Factory() {
+  protected ReconnectingChannel createReconnectingChannel(final InetAddress host) {
+    return new ReconnectingChannel( options.getTimeoutMs(), new ReconnectingChannel.Factory() {
+      @SuppressWarnings("deprecation")
       @Override
       public Channel createChannel() {
-        return NettyChannelBuilder
-            .forAddress(
-              new InetSocketAddress(transportOptions.getHost(), transportOptions.getPort()))
-            .sslContext(transportOptions.createSslContext())
-            .eventLoopGroup(transportOptions.getEventLoopGroup()).executor(executor)
-            .negotiationType(NegotiationType.TLS).streamWindowSize(1 << 20) // 1 MB
-            .build();
+        try {
+          return NettyChannelBuilder
+              .forAddress(new InetSocketAddress(host, options.getPort()))
+              .sslContext(SslContext.newClientContext())
+              .eventLoopGroup(elg)
+              .executor(batchPool)
+              .negotiationType(NegotiationType.TLS)
+              .streamWindowSize(1 << 20) // 1 MB
+              .build();
+        } catch (SSLException e) {
```

Comment:
Seems SslContext.newClientContext throws an IllegalStateException already, so maybe the SSLException catch is not needed.